File: frontend/src/main/frontend/src/app/events/event.ts
import {Profile} from "../account/profile";
export class Evento{
  eventId : number;
  name : string;
  eventDate : string;
  description : string;
  periodicityId : number;
  ownerLogin: string;
  periodicity : string;
  place : string;
  eventTypeId : number;
  eventType : string;
  isDraft : boolean;
  folderId : number;
  imageFilepath : string;
  ownerId : string;
  participants : Profile[];

  clone() : Evento {
    let itemClone = new Evento();

    itemClone.eventId = this.eventId;
    itemClone.name = this.name;
    itemClone.eventDate = this.eventDate;
    itemClone.periodicity = this.periodicity;
    itemClone.place = this.place;
    itemClone.eventTypeId = this.eventTypeId;
    itemClone.eventType = this.eventType;
    itemClone.isDraft = this.isDraft;
    itemClone.folderId = this.folderId;
    itemClone.ownerId = this.ownerId;
    itemClone.imageFilepath = this.imageFilepath;
    itemClone.imageFilepath = this.imageFilepath;
    itemClone.participants = this.participants;

    return itemClone;
  }
}

File: third-service/src/main/java/com/kpi/project/third/service/entity/Event.java
package com.kpi.project.third.service.entity;


import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Max;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Event {

    private int eventId;

    @NotBlank
    @Size(min = 4, max = 50)
    private String name;

    private String eventDate;
    private String pinnedEvent;

    @Size(max = 1023)
    private String description;

    private int periodicityId;
    private EventPeriodicity periodicity;

    @Size(min = 4, max = 100)
    private String place;
    private int eventTypeId;
    private EventType eventType;

    @JsonProperty()
    private boolean isDraft;

    @NotNull
    private int folderId;
    private String imageFilepath;
    private int ownerId;

    private List<User> participants;

    public boolean getIsDraft() {
        return isDraft;
    }

    public void setIsDraft(boolean isDraft) {
        this.isDraft = isDraft;
    }
}
